server/libs/domain/src/media/media.service.ts

```diff
@@ -9,6 +9,8 @@ import { ISystemConfigRepository, SystemConfigFFmpegDto } from '../system-config
 import { SystemConfigCore } from '../system-config/system-config.core';
 import { JPEG_THUMBNAIL_SIZE, WEBP_THUMBNAIL_SIZE } from './media.constant';
 import { AudioStreamInfo, IMediaRepository, VideoStreamInfo } from './media.repository';
+import { rgbaToThumbHash } from 'thumbhash-node';
+import sharp from "sharp";
```

Comment:
The code in "domain" is supposed to be as technology and library independent as possible. `thumbhash-node` and `sharp` are both specific implementations that are unrelated to the "business logic". 

Can you instead:
- Add an interface call for this in the `domain/`
- Add the implementation of the method in the `infra/`

I think it make sense to use the `media.repository.ts` which lives:
- `libs/domain/src/media/media.repository.ts`
- `libs/infra/src/media/repositories/media.repository.ts`

You can look at other examples in the `media.service.ts` / `media.repositroy.ts`, like for thumbnail generation.

I think it makes sense for the method to look something like this:

```
interface ThumbhashOptions {
  width: number;
  height: number;
}
getThumbhash(input: string, options: ThumbhashOptions): Promise<Buffer>;
```